### DISCUSSION TITLE
Multi-dimensional notification getting notification when on alert has no data

### DISCUSSION THREAD

**Discussion by yzaltz** [grafana/grafana#67614] [Alerting]
Using Grafana 9.4.7
I have several queries - measuring action speed.
trying to send notification when speed is below a threshold or there is no data.

when I have data on all queries - it works as a charm
but when one of the queries returns no data I get an empty notification.

This is the AlertValues annotation.

{{ with $values }}
{{ range $k, $v := . }}
{{if $v.Labels.action}}
Action: {{$v.Labels.action}} : {{ $v }}
{{end}}
{{if $v.Labels.service}}
Service: {{$v.Labels.service}} : {{ $v }}
{{end}}
{{ end }}
{{ end }}

When all queries return data I get something like:
Action: :
Action: :

What I would like to get when one or more of the queries return no data (while others return data) is something like :

Action: :
Action: : No Data

Kindly advise

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is phrased as a question, and we'd like questions about grafana to go to our[ community forum](https://community.grafana.com/). Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
